scripts/get_indexfiles.py

```diff
@@ -2,88 +2,134 @@
-
-def download_index(query: str, output_dir: str, params: str, log_file: str,
-                   mode: str = 'w'):
-    """Calls the script :program:`cdx-index-client.py` to do the actual work."""
-    with open(log_file, '{}t'.format(mode)) as logf:
-        res = subprocess.run(
-            'cdx-index-client.py --fl url,filename,offset,length,status,mime '
-            '-z {} -d {} {}'.format(query, output_dir, params),
-            stdout=logf, stderr=subprocess.STDOUT, shell=True, check=True
-        )
-        return res.returncode
-
-
-def get_uncompleted(log_file: str) -> Dict[str, List[str]]:
-    """Returns the list of """
-    mrp = re.compile(r'^Max retries .* page (?P<page>[0-9any]+) '
-                     r'for crawl (?P<coll>[A-Z0-9]+)-index$')
-    with open(log_file, 'rt') as inf:
-        colls = defaultdict(list)
-        for line in map(str.strip, inf):
-            m = mrp.match(line)
-            if m:
-                colls[m.group('coll')].append(m.group('page'))
-    return colls
+    parser.add_argument('--log-level', '-L', type=str, default='info',
+                        choices=['debug', 'info', 'warning', 'error', 'critical'],
+                        help='the logging level.')
+    args = parser.parse_args()
+
+    args.field_list = {field.strip() for field in args.field_list.split(',')}
+    return args
 
 
 def main():
     args = parse_arguments()
-    i = 0
-    log_file = args.log_file + '.{}'.format(i)
-    rcode = download_index(args.query, args.output_dir,
-                           '-c {}'.format(args.collection), log_file)
-    while rcode == 0 and i <= args.max_retry:
-        uncompleted = get_uncompleted(log_file)
-        if not uncompleted:
-            break
-        i += 1
-        log_file = args.log_file + '.{}'.format(i)
-        print('Doing {}th round of full retry'.format(i), file=sys.stderr)
-        if os.path.exists(log_file):
-            os.remove(log_file)
-        for coll, pages in uncompleted:
-            time.sleep(30)  # Sleep to prevent DDoS
-            download_index(args.query, args.output_dir,
-                           '-c {} --pages {}'.format(coll, ' '.join(pages)),
-                           log_file, 'a')
-
-    if i <= args.max_retry:
-        print('Successfully finished after {} iterations!'.format(i + 1),
-              file=sys.stderr)
+
+    logging.basicConfig(
+        level=getattr(logging, args.log_level.upper()),
+        format='%(asctime)s - %(process)s - %(levelname)s - %(message)s'
+    )
+
+    base_url = 'https://data.commoncrawl.org/cc-index/collections/' \
+               f'{args.collection}/indexes/'
+
+    if args.clusters_dir:
+        clusters_dir = args.clusters_dir
+        logging.info(f'Using clusters directory {clusters_dir}.')
     else:
-        print('Finished after {} of {} iterations please check if everything '
-              'is all right!'.format(i + 1, args.max_retry + 1), file=sys.stderr)
+        clusters_dir = Path(mkdtemp())
```

Comment:
What is the reason for not using the higher level TemporaryDirectory() method, which cleans up itself?

And why are you casting it to Path? I thought it returns a Path object by default.